pandas/tests/window/test_rolling.py

```diff
@@ -326,3 +326,10 @@ def test_rolling_axis_count(self, axis_frame):
 
         result = df.rolling(2, axis=axis_frame).count()
         tm.assert_frame_equal(result, expected)
+
+    def test_readonly_array(self):
+        arr = np.array([1, 3, np.nan, 3, 5])
+        arr.setflags(write=False)
+        tm.assert_series_equal(
+            pd.Series(arr).rolling(2).mean(), pd.Series([np.nan, 2, np.nan, np.nan, 4])
```

Comment:
Just assign the first argument to `result` and the second to `expected` and do tm.assert_series_equal(result, expected) (standard in tests)